Add functionality to expand journey on /add page to see all invidual sections

mocherfaoui opened the pull request:
will close #60

still work in progress. 
I see that there is a desktop and mobile components? why not combine them in one component?

let me know what you think, and if there are anything to change/add

noahflk commented:
Thank you @mocherfaoui. I really like it!

>  I see that there is a desktop and mobile components? why not combine them in one component?

I did that because the alignment/styling of each journey result differs slighly between mobile and desktop. I found it easier to create two components vs adjusting the layout with CSS. But feel free to combine it into one if that makes showing the detail view on mobile easier.

**Two things I'd tweak. First:**

You did a great job copying the "Train change" label from sbb.ch. But I think the following styling would work better for the Railtrack design language:

```tsx
<div className="my-8 flex space-x-8 rounded bg-gray-100 py-3 px-6">
  <RefreshIcon className="h-6 w-6 text-gray-600" />
  <p className="text-sm text-gray-600">Train change</p>
</div>
```

What do you think?

**Second:**

The stop indicator bullets aren't that aligned:
![grafik](https://user-images.githubusercontent.com/82475733/208108318-fd82a1f2-6433-4199-8fe4-01b60902333f.png)

You could look at how I did this in `JourneyStopIndicator.tsx`. This might help. Also, I think we should change the color from `black` to `primary` (dark red).

Great work. Can't wait to get this merged 💪

mocherfaoui commented:
@noahflk thanks for reviewing

> You did a great job copying the "Train change" label from sbb.ch. But I think the following styling would work better for the Railtrack design language

I agree with this, I changed it accordingly

I fixed the alignment issue, it was related to the font weight somehow. and I totally forgot about color and used black by default 😅 
for the mobile, I may agree with you about using a separate component for it, the only issue is -like in this case- I need to implement same feature for both desktop/mobile which can be daunting if you want to fix it since it's like a double implementation if you get what I mean

noahflk commented:
@mocherfaoui merged, thank you very much! 🎉

> there is only one feature I want to implement next. A button to get the extended view of all stations in a trip, since some have like 11 pass list so the expanded details can be long. what do you think?

I agree, that would be useful. Feel free to submit another PR for this 👌 But it's not a pressing matter. So I have no issues releasing this as-is now.
